RecordCollection/Controllers/AlbumsController.cs

```diff
@@ -15,8 +19,39 @@ public AlbumsController(RecordCollectionContext context)
         public IActionResult Index()
         {
             var albums = _context.Albums.ToList();
-            
             return View(albums);
         }
+
+        [Route("Albums/{id:int}")]
+        public async Task<IActionResult> Show(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var retrieved = await _context.Albums.FindAsync(id);
+
+            var relatedAlbums = _context.Albums
+                .Where(a => a.Artist == retrieved.Artist && a.Id != retrieved.Id)
+                .OrderBy(a => a.ReleaseDate);
```

Comment:
Additionally, I would recommend using the extract method again to create a method to retrieve related albums. As this code will likely need to be run multiple times throughout the program, this would adhere to the DRY principle.